content/upcoming/announcing_babelfish.md

```diff
@@ -0,0 +1,86 @@
+---
+author: abeaumont
+date: 2017-06-06
+title: "Announcing Babelfish"
+draft: true
+image: **TODO: Babelfish image (/post/announcing-babelfish/babelfish.jpg)**
+description: "Announcing Babelfish, the project we are developing to build
+representations of source code."
+categories: ["technical"]
+---
+
+At source{d} we believe there's a possibility for programs to write their own
+code and for new forms of automatic programming to emerge. Our
+[Roadmap](https://blog.sourced.tech/post/our-roadmap/) states the first step in this
+direction as: *Build representations of source code, developers and projects.*
+
+Today we are announcing [Babelfish](https://doc.bblf.sh/), the project we are
+developing to build this representations of source code.
+
+## What's Babelfish?
+
+Babelfish is a universal code parser. It can parse any file, in any language,
+extract an abstract syntax tree (AST) from it, and convert it to a Universal
+Abstract Syntax Tree [(UAST)](https://doc.bblf.sh/uast/specification.html)
+
+Well, that's our objective, we are not there just yet, but we have designed an
+architecture to achieve exactly that, and we're currently working it.
+
+## The architecture of Babelfish
+
+**TODO: Architecture image**
+
+As shown in the figure, Babelfish is a client/server system. Clients are source
+code analysis tools that rely on the server for actual source code parsing,
+written in different programming languages.
+
+The server itself uses language drivers to perform the parsing, which are
+divided in two parts, in order to minimise the work of developing a new driver:
+
+- A language code parser, which builds a native AST. This parser can be built
+  directly from target language's compiler tools or libraries.
+- An AST normalizer written in Go, that gives the UAST as a result.
+
+The server uses containers to run these drivers through libcontainer. This
+frees the user from having to handle dozens of different language ecosystems,
+since the server just handles driver using Docker images.
+
+You can look at the [documentation](https://doc.bblf.sh/architecture.html) for
```

Comment:
Super minor: "You can look at the documentation..." sounds a little more RTFMish and less inviting than "Take a look at the documentation..." but this is super subjective.